foundation/plugins/org.jboss.tools.foundation.core/src/org/jboss/tools/foundation/core/credentials/internal/CredentialDomain.java

```diff
@@ -131,14 +167,17 @@ void saveToPreferences(Preferences prefs, ISecurePreferences securePrefs) throws
 			ISecurePreferences userNode = securePrefs.node(user);
 			userNode.put(PROPERTY_PASS, getCredentials(user), true);
 		}
+		
+		String[] promptedUsers = (String[]) promptedCredentials.toArray(new String[promptedCredentials.size()]);
```

Comment:
unnecessary array. you should just be able to String.join("\n", promptedCredentials)